libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublishingWorker.kt

```diff
@@ -0,0 +1,106 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.analytics
+
+import android.content.Context
+import androidx.work.ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE
+import androidx.work.ListenableWorker.Result.success
+import androidx.work.PeriodicWorkRequest.Builder
+import androidx.work.WorkManager.getInstance
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.salesforce.androidsdk.accounts.UserAccountManager
+import com.salesforce.androidsdk.analytics.AnalyticsPublishingWorker.Companion.reEnqueueAnalyticsPublishPeriodicWorkRequest
+import java.util.concurrent.TimeUnit.HOURS
+
+/**
+ * An Android background tasks worker which publishes stored analytics.
+ * This class is intended to be instantiated by the background tasks work
+ * manager.
+ *
+ * Use [reEnqueueAnalyticsPublishPeriodicWorkRequest] to enqueue a analytics
+ * publish periodic work request.  Previous requests will be cancelled.
+ *
+ * @param context The Android context provided by the work manager
+ * @param workerParams The worker parameters provided by the work manager
+ * @see <a href='https://developer.android.com/guide/background'>Android
+ * Background Tasks</a>
+ */
+internal class AnalyticsPublishingWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(
+    context,
+    workerParams
+) {
+
+    override fun doWork() =
+
+        // Publish all stored analytics for the current user.
+        UserAccountManager
+            .getInstance()
+            .cachedCurrentUser
+            .let { userAccount ->
+
+                SalesforceAnalyticsManager
+                    .getInstance(userAccount ?: return@let null)
+                    .publishAllEvents()
+            }.run { success() /* Though not having a user cancels analytics publishing, it's not a failure. */ }
+
+    companion object {
+
+        /**
+         * Enqueues a persistent background tasks periodic work request to
+         * publish stored analytics for the current user at the provided
+         * interval.
+         *
+         * If a work request is already queued, it will be cancelled before
+         * the replacement is enqueued.
+         *
+         * @param context The Android context
+         * @param publishHoursInterval The interval at which to publish in hours
+         * @return UUID The worker's unique id, which may be used for
+         * cancellation
+         */
+        fun reEnqueueAnalyticsPublishPeriodicWorkRequest(
+            context: Context,
+            publishHoursInterval: Long
+        ) = Builder(
+            AnalyticsPublishingWorker::class.java,
+            publishHoursInterval,
+            HOURS
+        ).build().also { publishAnalyticsPeriodicWorkRequest ->
+            runCatching {
+                getInstance(context)
+            }.getOrNull()?.enqueueUniquePeriodicWork(
+                "SalesforceAnalyticsPublishingPeriodicWork",
```

Comment:
NIT:  Probably unnecessary but should this be a const on the companion in case it needs to get referenced again?

Reply:
I was on the verge of doing that, though for cancellation this provides the UUID of the unique job and that's probably what code outside this source would always use.  I made the new constant private to reflect that so cancellation is more likely to be done by id.